prow/tide/tide.go

```diff
@@ -234,6 +237,38 @@ func accumulate(presubmits []string, prs []pullRequest, pjs []kube.ProwJob) (suc
 	return
 }
 
+func prNumbers(prs []pullRequest) []int {
+	var nums []int
+	for _, pr := range prs {
+		nums = append(nums, int(pr.Number))
+	}
+	return nums
+}
+
+func (c *Controller) pickBatch(sp subpool) ([]pullRequest, error) {
+	r, err := c.gc.Clone(sp.org + "/" + sp.repo)
+	if err != nil {
+		return nil, err
+	}
+	defer r.Clean()
+	if err := r.Checkout(sp.sha); err != nil {
+		return nil, err
+	}
+	var res []pullRequest
+	for _, pr := range sp.prs {
+		// TODO(spxtr): Check the actual statuses for individual jobs.
+		if string(pr.Commits.Nodes[0].Commit.Status.State) != "SUCCESS" {
+			continue
+		}
+		if ok, err := r.Merge(string(pr.HeadRef.Target.OID)); err != nil {
```

Comment:
Don't you need to fetch the PR heads?

Reply:
Nope, it's a mirror clone so it has all that stuff.